rust/src/coordinator_async/sum.rs

```diff
@@ -0,0 +1,111 @@
+use super::{State, StateMachine};
+use crate::{
+    coordinator_async::{
+        error::Error,
+        message_processing::{MessageHandler, MessageSink, SumValidationData},
+        update::Update,
+    },
+    message::{MessageOwned, PayloadOwned},
+    PetError,
+};
+use std::{collections::HashMap, default::Default, future::Future, pin::Pin, sync::Arc};
+use tokio::{
+    sync::{broadcast, mpsc},
+    time::Duration,
+};
+
+#[derive(Debug)]
+pub struct Sum;
+
+impl State<Sum> {
+    pub async fn next(mut self) -> StateMachine {
+        println!("Sum phase!");
+
+        match self.run().await {
+            Ok(_) => {
+                // Fetch sum dict?
+                let sum_dict = HashMap::new();
+                StateMachine::Update(State {
+                    _inner: Update {
+                        sum_dict: Some(Arc::new(sum_dict)),
+                    },
+                    coordinator_state: self.coordinator_state,
+                    message_rx: self.message_rx,
+                })
+            }
+
+            Err(_) => StateMachine::Error(State {
+                _inner: Error {},
+                coordinator_state: self.coordinator_state,
+                message_rx: self.message_rx,
+            }),
+        }
+    }
+
+    async fn run(&mut self) -> Result<(), PetError> {
```

Comment:
all the cloning in this method seems quite wasteful, is this because of the aforementioned borrow checker issues for the `select!` macro?